### PR TITLE
Survey inputs in templates (issue 9)

### PR DESCRIPTION
**Pull request by maxmeister**
Dear Friends of the Deploji project,

I think the feature described in issue 9 is ready. Please conduct a code review, make sure the code adheres to the highest standards of quality and check if the business requirements are met.

### CODE REVIEW

File: src/app/scam/components/templates/survey-send/survey-send.component.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          <mat-form-field class="w-100">
+            <mat-label>{{input.Label}}</mat-label>
+            <textarea matInput [placeholder]="input.Hint" formControlName="{{i}}"></textarea>
+          </mat-form-field>
+        </div>
+        <div *ngSwitchCase="surveyTypes.SELECT">
+          <mat-form-field class="w-100">
+            <mat-label>{{input.Label}}</mat-label>
+            <mat-select formControlName="{{i}}">
+              <mat-option *ngFor="let item of selectItems(input.Hint)" [value]="item">
```
Comment (sotomskir) [inline src/app/scam/components/templates/survey-send/survey-send.component.html]: change `selectItems(input.Hint)` to `selectItems(input.Options)`

File: src/app/scam/components/templates/surveys/surveys.component.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        <input matInput type="text" formControlName="VariableName">
+      </mat-form-field>
+      <mat-form-field>
+        <mat-label>Type</mat-label>
+        <mat-select formControlName="Type">
+          <mat-option *ngFor="let type of types" [value]="type">
+            {{type}}
+          </mat-option>
+        </mat-select>
+      </mat-form-field>
```
Comment (sotomskir) [inline src/app/scam/components/templates/surveys/surveys.component.html]: When the user selects input of type "select" an additional textarea for select options should appear below. 
With `formControlName="Options"`

File: src/app/scam/components/shared/template-launch/template-launch.component.ts [hunk 3]
```diff
@@ -14,39 +14,60 @@ import { FormSshKeyComponentModule } from '../form/form-ssh-key/form-ssh-key.com
...
+    private jobsService: JobsService,
+    private router: Router,
+    private surveyService: SurveyService,
+    private surveyInputManager: SurveyInputManagerService
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.surveyInputManager.inputSource.subscribe((extraVariables: string) => {
+      this.form.ExtraVariables.setValue(extraVariables);
```
Comment (sotomskir) [inline src/app/scam/components/shared/template-launch/template-launch.component.ts]: I think extra variables from the survey shouldn't overwrite original extra variables. It should be merged.

File: src/app/scam/components/shared/template-launch/template-launch.component.html [hunk 4]
```diff
@@ -5,10 +5,7 @@ <h3>Launch template {{template?.Name}}</h3>
     <app-form-inventory *ngIf="template?.PromptInventory" [control]="form.Inventory"></app-form-inventory>
     <app-form-project-file *ngIf="template?.PromptPlaybook" [control]="form.Playbook" label="Playbook" [project]="form.get('Project').value"></app-form-project-file>
     <app-form-ssh-key *ngIf="template?.PromptSshKey" [control]="form.SshKey"></app-form-ssh-key>
-    <mat-form-field *ngIf="template?.PromptExtraVariables" class="block">
-      <mat-label>Extra variables</mat-label>
-      <textarea cdkTextareaAutosize matInput type="textarea" formControlName="ExtraVariables"></textarea>
-    </mat-form-field>
```
Comment (sotomskir) [inline src/app/scam/components/shared/template-launch/template-launch.component.html]: I think that this should be kept, and merged with survey inputs when the user click "Launch"

**Comment by maxmeister** [comment]
Thank you for a thorough review and pointing out bugs. Will send a commit fixing those shortly.

File: src/app/scam/components/templates/surveys/surveys.component.html [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                <mat-icon (click)="editExtraVariable(i, item)" class="icon">edit</mat-icon>
+                <mat-icon (click)="deleteExtraVariable(i, item)" class="icon icon--delete">delete</mat-icon>
+              </div>
+            </div>
+          </div>
+          <div *ngSwitchCase="surveyTypes.SELECT">
+            <mat-form-field class="w-84">
+              <mat-label>{{item.Label}}</mat-label>
+              <mat-select>
+                <mat-option *ngFor="let option of parseSelectItems(item.Hint)" [value]="option">
```
Comment (sotomskir) [inline src/app/scam/components/templates/surveys/surveys.component.html]: Please change item.Hint to item.Options

File: src/app/scam/components/templates/surveys/surveys.component.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+export class SurveysComponent implements OnInit {
+
+  public types: string[] = [SurveyInputTypes.TEXT, SurveyInputTypes.SELECT, SurveyInputTypes.TEXTAREA];
+  public surveyTypes = SurveyInputTypes;
+  public formList: SurveyListForm = new SurveyListForm();
+  public formDetails: SurveyDetailsForm = new SurveyDetailsForm();
+  public templateId: number;
+  public survey: Survey;
+  private surveyIdToEdit: number = null;
+  private subscription: Subscription = new Subscription();
```
Comment (sotomskir) [inline src/app/scam/components/templates/surveys/surveys.component.ts]: unsubscribe in OnDestroy

File: src/app/scam/components/templates/surveys/surveys.component.html (hunk 5 above)
Comment (maxmeister) [reply src/app/scam/components/templates/surveys/surveys.component.html]: Done

File: src/app/scam/components/templates/surveys/surveys.component.ts (hunk 6 above)
Comment (maxmeister) [reply src/app/scam/components/templates/surveys/surveys.component.ts]: Done

**Comment by sotomskir** [comment]
@maxmeister Thank You for contributing to Deploji project. This change will be released in version 1.5.0

**Pull request merged**